src/custom/state/orders/hooks.ts

```diff
@@ -72,17 +72,14 @@ type UpdateLastCheckedBlockCallback = (updateLastCheckedBlockParams: UpdateLastC
 
 type GetOrderByIdCallback = (id: OrderID) => SerializedOrder | undefined
 
-type OrderTypeKeys = 'pending' | 'expired' | 'fulfilled' | 'cancelled'
-function _concatOrdersState(state: OrdersState[ChainId], keys: OrderTypeKeys[]) {
+function _concatOrdersState(state: OrdersState[ChainId]) {
+  const keys = mapEnumKeysToArray<OrderTypeKeys>(OrderTypeKeys)
   if (!state) return []
 
-  const firstState = state[keys[0]] || {}
-  const restKeys = keys.slice(1)
-
-  return restKeys.reduce((acc, nextKey) => {
+  return keys.reduce((acc, nextKey) => {
     const nextState = Object.values(state[nextKey] || {})
     return acc.concat(nextState)
-  }, Object.values(firstState))
+  }, [] as (OrderObject | undefined)[])
```

Comment:
Is not this the same as a just using a spread operator with the `oldOrder` and 

```js
{ pending: [], expired: [], 'fulfilled: [], cancelled': [] }
```

Also, I would love if u add some comments. For me was not obvious what u tried to accomplish and why. Nice if u write that for our future-selfs or other people. I understand has to do with the order having less properties because it's recovered from a persisted order from an old version. If that's the case, it's relevant enough to explain